### PR TITLE
fix: Remove first_release constraint on Group

### PR DESCRIPTION
**Pull request by dcramer**
Rename Group.first_release to Group.first_release_id and update appropriate code paths.

Includes a dangerous migration to drop the foreign key.

This came up do to project transfer not fully managing these constraints, which means we have some invalid rows blocked on constraints right now.

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -1243,16 +1243,14 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
...
 
             with transaction.atomic():
                 short_id = project.next_short_id()
                 group, group_is_new = Group.objects.create(
                     project=project,
                     short_id=short_id,
-                    first_release_id=Release.objects.filter(
-                        id=first_release.id,
-                    ).values_list('id', flat=True).first() if first_release else None,
+                    first_release_id=first_release_id,
```
Comment (markstory) [inline src/sentry/event_manager.py]: Nice to save a query here :+1:

**Review by markstory — COMMENTED** [review]
Do you think any plugins would be leveraging the association that is being removed?

**Comment by dcramer** [comment]
@markstory nothing that we’ve ever built should be using it

**Comment by dcramer** [comment]
@markstory thoughts on if we should still try to get this done?

**Comment by markstory** [comment]
When a project is transferred are releases also transferred? Would potentially be introducing cross organization references with this change?

**Comment by BYK** [comment]
Closing due to staleness and conflicts.

**Pull request closed without merging**